clarifai/client/base_auth.py

```diff
@@ -0,0 +1,30 @@
+from clarifai.client.auth import create_stub
+from clarifai.client.auth.stub import ClarifaiAuthHelper
+
+
+class BaseAuth(object):
+  """
+    BaseAuth is the base class for all the classes interacting with Clarifai endpoints.
+
+    Parameters:
+        **kwargs: Additional keyword arguments to be passed to the ClarifaiAuthHelper.
+            - user_id (str): A user ID for authentication.
+            - app_id (str): An app ID for the application to interact with.
+            - pat (str): A personal access token for authentication.
+            - base (str): The base URL for the API endpoint. Defaults to 'https://api.clarifai.com'.
+            - ui (str): The URL for the UI. Defaults to 'https://clarifai.com'.
+
+    Attributes:
+        auth_helper (ClarifaiAuthHelper): An instance of ClarifaiAuthHelper for authentication.
+        STUB (Stub): The gRPC Stub object for API interaction.
+        metadata (tuple): The gRPC metadata containing the personal access token.
+        userDataObject (UserAppIDSet): The protobuf object representing user and app IDs.
+        base (str): The base URL for the API endpoint.
+    """
+
+  def __init__(self, **kwargs):
+    self.auth_helper = ClarifaiAuthHelper(**kwargs)
+    self.STUB = create_stub(self.auth_helper)
+    self.metadata = self.auth_helper.metadata
+    self.userDataObject = self.auth_helper.get_user_app_id_proto()
+    self.base = self.auth_helper.base
```

Comment:
Is the `BaseAuth` class really needed? 
It looks like everything it provides is already given in `ClarifaiAuthHelper()`

Reply:
By using `BaseAuth`, users are shielded from the complexities of the authhelper. Additionally, it allows for the centralized usage of create_stub across various class interfaces such as inputs, datasets, models, modules, and more

Reply:
Right, I agree with the centralized usage of `create_stub()`.

On that note, how will users know what keyword args to pass to the class?

Reply:
The `BaseAuth` is designed to remain hidden from users. Instead, users are encouraged to set the `PAT` (Personal Access Token) as an environment variable at the beginning. The `kwargs` here internally pass `user_id` and `app_id` to obtain the `userDataObject` in other classes.